client/components/FormattingBar/controlComponents/ControlsReference.tsx

```diff
@@ -26,58 +22,33 @@ const ControlsReference = (props: ControlsReferenceProps) => {
 		editorChangeObject: { updateNode, selectedNode, view },
 	} = props;
 	const { blockNames } = usePubContext();
-
-	const possibleTargets = useMemo(() => getReferenceableNodes(view.state), [
-		view.state,
-		blockNames,
-	]);
+	const possibleTargets = useMemo(() => getReferenceableNodes(view.state), [view.state]);
 	const [target, setTarget] = useState(() => matchInitialTarget(selectedNode, possibleTargets));
 	const targetId = target && target.node && target.node.attrs.id;
 	const changed = useRef(false);
 
-	const currentIcon = target ? target.icon : 'disable';
-	const currentLabel = target
-		? buildLabel(target.node, blockNames[target.node.type.name])
-		: possibleTargets.length
-		? 'No referenced item'
-		: 'No items to reference';
-
 	useEffect(() => {
 		if (changed.current) {
 			updateNode({ targetId: targetId });
 			changed.current = false;
 		}
 	}, [targetId, updateNode]);
 
+	const onSelect = useCallback(
+		(reference: NodeReference) => {
+			changed.current = true;
+			setTarget(reference);
+		},
+		[setTarget],
+	);
+
 	return (
-		<div className="controls-link-component">
-			<MenuButton
-				disabled={possibleTargets.length === 0}
-				aria-label="Select an item to reference"
-				buttonContent={currentLabel}
-				buttonProps={{
-					rightIcon: 'chevron-down',
-					minimal: true,
-					icon: currentIcon,
-				}}
-			>
-				{possibleTargets.map((possibleTarget) => {
-					const { icon, node } = possibleTarget;
-					return (
-						<MenuItem
-							onClick={() => {
-								changed.current = true;
-								setTarget(possibleTarget);
-							}}
-							key={node.attrs.id}
-							active={targetId && targetId === node.attrs.id}
-							text={buildLabel(node, blockNames[node.type.name])}
-							icon={icon}
-						/>
-					);
-				})}
-			</MenuButton>
-		</div>
+		<ReferencesDropdown
+			references={possibleTargets}
+			selectedReference={target}
+			blockNames={blockNames}
+			onSelect={onSelect}
+		/>
```

Comment:
I broke this component out with the intention of using it for the dropdown, but then realized the suggestion dropdown didn't need `MenuButton` behavior. I left it broken out into its own component, anyways. Thoughts on merging it back in?